ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlRestApplication.java

```diff
@@ -646,7 +668,8 @@ private static void maybeCreateProcessingLogStream(
       return;
     }
 
-    commandQueue.enqueueCommand(configured.get());
+    commandQueue.enqueueCommand(configured.get(), transactionalProducer);
+    transactionalProducer.commit();
```

Comment:
is the exception handling and the abortTransaction() handled inside `enqueueCommand` > `Transactional Producer::send`? could we avoid calling commit() for an aborted transaction ?